### PR TITLE
Update type documentations for #16307, #16359

### LINKED ISSUE #16307: TypeScript integration issue due to wrong declarations
Link to TypeScript file here:
https://github.com/mozilla/pdfjs-dist/blob/5f07d5a4159bb99eee2f6143d1297f03b45bba58/types/web/pdf_find_controller.d.ts#L56

Configuration:
- PDF.js version: 3.5.141
- TypeScript version: 5.0.4

The following generated TypeScript declaration prevents calling of PDF.js API.
https://github.com/mozilla/pdfjs-dist/blob/5f07d5a4159bb99eee2f6143d1297f03b45bba58/types/web/pdf_find_controller.d.ts#L56
```
element?: null | undefined;
```
Here the property element is restricted to the types null or undefined.

The **pdf_find_controller.d.ts** declaration file is generated by 
https://github.com/mozilla/pdf.js/blob/3e08eee511b04e127fcd41d2a1d07dd3b5a68a92/gulpfile.js#L1514

A possible fix would be to declare the type like:
```typescript
element?: any;
```
This would be the case when calling TSC without *--strict* option:
https://github.com/mozilla/pdf.js/blob/3e08eee511b04e127fcd41d2a1d07dd3b5a68a92/gulpfile.js#L1507
```javascript
// Remove --strict option and generate type declarations with
gulp types
````

Example snippet which uses the PDF.js TypeScript API:
```typescript
import { EventBus, PDFLinkService, PDFFindController } from "pdfjs-dist/web/pdf_viewer";

const pdfFindController = new PDFFindController({
	eventBus: new EventBus(),
	linkService: new PDFLinkService(),
	updateMatchesCountOnProgress: true
});

// TSC compiles TypeScript code successful
pdfFindController.scrollMatchIntoView({ element: null });
pdfFindController.scrollMatchIntoView({ element: undefined });

// TSC compilation error due to wrong declaration in pdf_find_controller.d.ts
pdfFindController.scrollMatchIntoView({ element: "non null" });
pdfFindController.scrollMatchIntoView({ element: document.getElementById("testId") });
```

API can't be called with a real object and leads to following compilation issues:
```
ERROR in snippet.ts(14,41)
      TS2322: Type 'string' is not assignable to type 'null'.

ERROR in snippet.ts(25,41)
      TS2322: Type 'HTMLElement' is not assignable to type 'null'.
```

Steps to reproduce the problem:
1. Compile above TypeScript snippet with TSC.
2. See compilation issues

What is the expected behavior? (add screenshot)
Calling of PDF.js TypeScript API is possible.

What went wrong? (add screenshot)
Compilaton of TypeScript code with TSC.

### LINKED ISSUE #16359: Bad typing for argument rotation of `PDFPageView.update`
PDF.js version: 3.5.141

The type of `rotation` property in argument of `PDFPageView.update` is marked as `null | undefined` but it should be `number | undefined`

### PR DESCRIPTION
**Pull request by SebCorbin**
Should fix the typing problems for these issues:

- #16307
- #16359
- #16360

### CODE REVIEW

File: web/pdf_find_controller.js [hunk 1]
```diff
@@ -517,6 +517,14 @@ class PDFFindController {
     });
   }
 
+  /**
+   * Updates scroll so that the match appears into view.
+   *
```
Comment (Snuffleupagus) [inline web/pdf_find_controller.js:522]:
```suggestion
   * Scroll the current match into view.
```

File: web/pdf_find_controller.js [hunk 2]
```diff
@@ -517,6 +517,14 @@ class PDFFindController {
...
+   * @param {HTMLElement} element
+   * @param {number} selectedLeft
+   * @param {number} pageIndex
+   * @param {number} matchIndex
+   */
```
Comment (Snuffleupagus) [inline web/pdf_find_controller.js:527]: Given that the method takes an *object*, surely this needs to be a `typedef` in order for it to work correctly!?

File: web/pdf_page_view.js [hunk 3]
```diff
@@ -513,6 +513,22 @@ class PDFPageView {
     }
   }
 
+  /**
+   * @typedef {Object} PDFPageViewUpdateParameters
+   *
```
Comment (Snuffleupagus) [inline web/pdf_page_view.js:518]: Nit: We usually don't bother with a blank line in these cases
```suggestion
   * @typedef {Object} PDFPageViewUpdateParameters
```

File: web/pdf_page_view.js [hunk 4]
```diff
@@ -513,6 +513,22 @@ class PDFPageView {
...
+   * @property {Promise<OptionalContentConfig>} [optionalContentConfigPromise]
+   *   A promise that is resolved with an {@link OptionalContentConfig} instance.
+   *   The default value is `null`.
+   * @property {number} [drawingDelay]
+   */
+
+  /**
+   * Update page view by eitheir modifiying scale or rotation.
+   *
+   * @param {PDFPageViewUpdateParameters}
```
Comment (Snuffleupagus) [inline web/pdf_page_view.js:530]: Note that this method does more than only changing the scale/rotation, hence my suggestion:
```suggestion
   * Update e.g. the scale and/or rotation of the page.
   * @param {PDFPageViewUpdateParameters}
```

File: web/pdf_find_controller.js (hunk 2 above)
Comment (SebCorbin) [reply web/pdf_find_controller.js]: Indeed, I mixed it up with `scrollIntoView`

**Comment by Snuffleupagus** [comment]
Please fix the linting failure, note that you can run `gulp lint` locally to find those.

Please also squash the commits, see https://github.com/mozilla/pdf.js/wiki/Squashing-Commits, and update the commit message to remove the already fixed issue from it.

File: web/pdf_page_view.js [hunk 5]
```diff
@@ -513,6 +513,20 @@ class PDFPageView {
     }
   }
 
+  /**
+   * @typedef {Object} PDFPageViewUpdateParameters
+   * @property {number} [scale] The new scale, if specified.
+   * @property {number} [rotation] The new rotation, if specified.
+   * @property {Promise<OptionalContentConfig>} [optionalContentConfigPromise]
+   *   A promise that is resolved with an {@link OptionalContentConfig} instance.
+   *   The default value is `null`.
```
Comment (Snuffleupagus) [inline web/pdf_page_view.js:522]: The following should fix the linting error:
```suggestion
   *   A promise that is resolved with an {@link OptionalContentConfig}
   *   instance. The default value is `null`.
```

**Review by Snuffleupagus — APPROVED** [review]
r=me, thank you!

**Pull request merged**